addons/web_tour/static/src/js/tour_service.js

```diff
@@ -33,25 +34,98 @@ return session.is_bound.then(function () {
         var consumed_tours = session.is_frontend ? results[0] : session.web_tours;
         var tour_manager = new TourManager(rootWidget, consumed_tours);
 
-        // Use a MutationObserver to detect DOM changes
-        var untracked_classnames = ["o_tooltip", "o_tooltip_content", "o_tooltip_overlay"];
-        var check_tooltip = _.debounce(function (records) {
-            var update = _.some(records, function (record) {
-                return !(is_untracked(record.target) ||
-                    _.some(record.addedNodes, is_untracked) ||
-                    _.some(record.removedNodes, is_untracked));
+        // Use a MutationObserver to detect DOM changes, when a mutation occurs,
+        // only add it to the list of mutation to process and delay the
+        // mutation processing. We have to record them all so that a change that
+        // is undo only a moment later is detected correctly. Most of them
+        // will trigger a tip check anyway so, most of the time, processing the
+        // first one will be enough to be sure that a tip update has to be done.
+        let mutationTimer = undefined;
+        let currentMutations = [];
+        const observer = new MutationObserver(records => {
+            clearTimeout(mutationTimer);
+            for (const record of records) {
+                const newValue = record.type === 'attributes'
+                    ? record.target.getAttribute(record.attributeName)
+                    : null;
+                currentMutations.push({
+                    record: record,
+                    oldValue: record.oldValue || '',
+                    newValue: newValue || '',
```

Comment:
why don't you only push the record, and to the computation later on?

Reply:
As discussed, let's try ;)